### PR TITLE
Correctly close cache managers and put temporary files in a temporary folder

### PR DESCRIPTION
**Pull request by henri-tremblay**

### CODE REVIEW

**Comment by ljacomet** [comment]
test this please

File: integration-test/src/test/java/org/ehcache/integration/transactions/xa/XACacheTest.java [hunk 1]
```diff
@@ -66,40 +67,61 @@
...
         .setServerId("XACacheTest")
         .setGracefulShutdownInterval(0);
   }
 
   @After
   public void tearDownBtm() throws Exception {
+    if(cacheManager != null) {
+      cacheManager.close();
+    }
+    transactionManager.shutdown();
```
Comment (ljacomet) [inline integration-test/src/test/java/org/ehcache/integration/transactions/xa/XACacheTest.java]: Adding this makes the following block useless? Although it seems the following block is _safer_ in a way.
@lorban Care to comment?

Comment (lorban) [reply integration-test/src/test/java/org/ehcache/integration/transactions/xa/XACacheTest.java]: Right, the `transactionManager.shutdown()` call is redundant with the `if ()` block below.

Comment (henri-tremblay) [reply integration-test/src/test/java/org/ehcache/integration/transactions/xa/XACacheTest.java]: Cool thanks

**Pull request merged**